In [1]:
import nltk

In [ ]:
nltk.download()

In [13]:
file_path = '/content/drive/MyDrive/CSE 419/Lab Assign-1/file_01.txt'

In [16]:
with open(file_path, 'r') as file:
    file_1 = file.read()

In [17]:
file_1

'Java Tutorial\nThis Java Tutorial is designed for beginners as well as experienced professionals. Whether you’re looking to learn the basics of Java or its advanced concepts, this free Java tutorial is the perfect resource for you.\n\nWhat is Java?\nDeveloped by Sun Microsystems in 1995, Java is a highly popular, object-oriented programming language. This platform independent programming language is utilized for Android development, web development, artificial intelligence, cloud applications, and much more.\n\nIn this tutorial, we will cover everything from the basics of Java syntax to advanced topics like object-oriented programming and exception handling. So, by the end of this tutorial, you will have a strong understanding of Java and be ready to start writing your own Java applications. So let’s get started on this comprehensive Java programming tutorial!'

In [19]:
sent = nltk.sent_tokenize(file_1)

In [20]:
sent

['Java Tutorial\nThis Java Tutorial is designed for beginners as well as experienced professionals.',
 'Whether you’re looking to learn the basics of Java or its advanced concepts, this free Java tutorial is the perfect resource for you.',
 'What is Java?',
 'Developed by Sun Microsystems in 1995, Java is a highly popular, object-oriented programming language.',
 'This platform independent programming language is utilized for Android development, web development, artificial intelligence, cloud applications, and much more.',
 'In this tutorial, we will cover everything from the basics of Java syntax to advanced topics like object-oriented programming and exception handling.',
 'So, by the end of this tutorial, you will have a strong understanding of Java and be ready to start writing your own Java applications.',
 'So let’s get started on this comprehensive Java programming tutorial!']

In [23]:
tokens_1 = nltk.word_tokenize(file_1)

In [24]:
tokens_1

['Java',
 'Tutorial',
 'This',
 'Java',
 'Tutorial',
 'is',
 'designed',
 'for',
 'beginners',
 'as',
 'well',
 'as',
 'experienced',
 'professionals',
 '.',
 'Whether',
 'you',
 '’',
 're',
 'looking',
 'to',
 'learn',
 'the',
 'basics',
 'of',
 'Java',
 'or',
 'its',
 'advanced',
 'concepts',
 ',',
 'this',
 'free',
 'Java',
 'tutorial',
 'is',
 'the',
 'perfect',
 'resource',
 'for',
 'you',
 '.',
 'What',
 'is',
 'Java',
 '?',
 'Developed',
 'by',
 'Sun',
 'Microsystems',
 'in',
 '1995',
 ',',
 'Java',
 'is',
 'a',
 'highly',
 'popular',
 ',',
 'object-oriented',
 'programming',
 'language',
 '.',
 'This',
 'platform',
 'independent',
 'programming',
 'language',
 'is',
 'utilized',
 'for',
 'Android',
 'development',
 ',',
 'web',
 'development',
 ',',
 'artificial',
 'intelligence',
 ',',
 'cloud',
 'applications',
 ',',
 'and',
 'much',
 'more',
 '.',
 'In',
 'this',
 'tutorial',
 ',',
 'we',
 'will',
 'cover',
 'everything',
 'from',
 'the',
 'basics',
 'of',
 'Java',
 'syntax'

In [25]:
token_count_f1 = len(tokens_1)

In [26]:
token_count_f1

152

In [43]:
from nltk.corpus import stopwords

In [44]:
nltk.download('stopwords')

[nltk_data] Downloading package stopwords to /root/nltk_data...
[nltk_data]   Package stopwords is already up-to-date!


True

In [45]:
stop_words = set(stopwords.words('english'))

In [50]:
tokens1_without_stopwords = [word for word in tokens_1 if word.lower() not in stop_words]
token_count_1 = len(tokens1_without_stopwords)
print(token_count_1)

96


In [29]:
file_path = '/content/drive/MyDrive/CSE 419/Lab Assign-1/file_02.txt'

with open(file_path, 'r') as file:
    file_2 = file.read()

In [32]:
tokens_2 = nltk.word_tokenize(file_2)
print(len(tokens_2))

295


In [51]:
tokens2_without_stopwords = [word for word in tokens_2 if word.lower() not in stop_words]
token_count_2 = len(tokens2_without_stopwords)
print(token_count_2)

188


In [34]:
file_path = '/content/drive/MyDrive/CSE 419/Lab Assign-1/file_03.txt'

with open(file_path, 'r') as file:
    file_3 = file.read()

In [35]:
tokens_3 = nltk.word_tokenize(file_3)
print(len(tokens_3))

476


In [52]:
tokens3_without_stopwords = [word for word in tokens_3 if word.lower() not in stop_words]
token_count_3 = len(tokens3_without_stopwords)
print(token_count_3)

304


In [36]:
file_path = '/content/drive/MyDrive/CSE 419/Lab Assign-1/file_04.txt'

with open(file_path, 'r') as file:
    file_4 = file.read()

In [37]:
tokens_4 = nltk.word_tokenize(file_4)
print(len(tokens_4))

1299


In [53]:
tokens4_without_stopwords = [word for word in tokens_4 if word.lower() not in stop_words]
token_count_4 = len(tokens4_without_stopwords)
print(token_count_4)

858


In [40]:
file_path = '/content/drive/MyDrive/CSE 419/Lab Assign-1/file_05.txt'

with open(file_path, 'r') as file:
    file_5 = file.read()

In [41]:
tokens_5 = nltk.word_tokenize(file_5)
print(len(tokens_5))

292


In [54]:
tokens5_without_stopwords = [word for word in tokens_5 if word.lower() not in stop_words]
token_count_5 = len(tokens5_without_stopwords)
print(token_count_5)

204


### Token Counts for Each File

- **file_01.txt**:
  - Tokens before removing stop-words: 152
  - Tokens after removing stop-words: 96

- **file_02.txt**:
  - Tokens before removing stop-words: 295
  - Tokens after removing stop-words: 188

- **file_03.txt**:
  - Tokens before removing stop-words: 476
  - Tokens after removing stop-words: 304

- **file_04.txt**:
  - Tokens before removing stop-words: 1299
  - Tokens after removing stop-words: 858

- **file_05.txt**:
  - Tokens before removing stop-words: 292
  - Tokens after removing stop-words: 204
